src/Data/Vec/Bounded/Base.agda

```diff
@@ -0,0 +1,82 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Bounded vectors, basic types and operations
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Vec.Bounded.Base where
+
+open import Level using (Level)
+open import Data.Nat.Base
+import Data.Nat.Properties as ℕₚ
+open import Data.Product using (_×_; _,_)
+open import Data.Vec.Base as Vec using (Vec)
+open import Data.These.Base as These using (These)
+open import Function
+open import Relation.Nullary
+open import Relation.Unary
+open import Relation.Binary.PropositionalEquality using (_≡_)
+
+private
+  variable
+    a b c p : Level
+    A : Set a
+    B : Set b
+    C : Set c
+
+------------------------------------------------------------------------
+-- Types
+
+infix 4 _,_
+record Bounded (A : Set a) (n : ℕ) : Set a where
```

Comment:
I think the type should probably be called `BoundedVec` rather than `Bounded` otherwise it's going to be a bit confusing if a user see it in the middle of a module.

Reply:
I find `BoundedVec` too long. :/
Basing the name on `Data.List.NonEmpty`, I suppose we could call this one `Vec≤`?

Reply:
I like `Vec≤` :smile: